integration_test/metadata/integration_metadata.go

```diff
@@ -132,7 +133,9 @@ func UnmarshalAndValidate(data []byte, i interface{}) error {
 
 func SliceContains(slice []string, toFind string) bool {
 	for _, entry := range slice {
-		if entry == toFind {
+		// vm.Platform (ex. `sles-15`) does not align with ImageSpec (ex. `suse-cloud:sles-15`)
+		// TODO, remove lingering references to "Platform" and use "Image Spec" instead.
+		if strings.HasSuffix(entry, toFind) {
```

Comment:
I am confident that SliceContains is not the right place to be working around this. SliceContains is just supposed to be the same as slices.Contains (https://pkg.go.dev/slices#Contains), recently introduced into the language. Looking at the 4 callsites to this function, 2 of them want to look for an exact match, and 2 of them have this problem where they are passing in vm.Platform and comparing that to a list of ImageSpecs. Instead of doing that, i think we should store an ImageSpec in the VM struct alongside the Platform and pass that in to metadata.SliceContains instead of vm.Platform. Longer term, we slowly migrate everything to refer to vm.ImageSpec and get rid of vm.Platform.

Reply:
Done